### PR TITLE
gomplate: init at 3.8.0

### PR DESCRIPTION
**Pull request by risicle**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by 06kellyjac** [comment]
I tried to make some suggestions using the built in code suggestions feature but it was pretty messy so I put them on another branch for you to look through:
https://github.com/06kellyjac/nixpkgs/commits/ris-gomplate-init

* gomplate: move vendorSha256 next to src
I prefer the `vendorSha256` next to `src` and I always see them together.

* gomplate: clean up meta
Can just do the `with stdenv.lib` on meta once to make things cleaner.

* gomplate: add version info
The `rev` was only useful in `src` so usually I'd move it in there, but it's actually useful for feeding gomplate it's version.
Sets `github.com/hairyhenderson/gomplate/v3/version.Version=v3.8.0` using existing values

image


Feel free to cherrypick these commits or copy/paste :slightly_smiling_face:

**Comment by risicle** [comment]
Thanks - I'm very much foreign to the golang corners of nix... will sneak some of those fixes. I turn off "allow modifications" because it really confuses me when people do that.

**Comment by risicle** [comment]
Oh dear I should have checked for existing PRs: #85157

@jlesquembre if you don't mind I might add you as a co-maintainer of this as you seemed to have interest before?

**Review by 06kellyjac — APPROVED** [review]
LGTM 👍
Probably best to get confirmation @jlesquembre  is still ok being a maintainer before merging.

**Comment by jlesquembre** [comment]
Writing from my phone, I didn't have time to check it, but ok to leave me as maintainer. Happy to see gomplate in nixpkgs

**Pull request merged**
